frontend/src/components/DataCard/datacard.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from "react";
+import { useInView } from "react-intersection-observer";
```

Comment:
I see we have installed a package but PR is not showing any changes in package.json or lock-file. Please install and push those changes as well

Reply:
Done.